lib/reline/unicode.rb

```diff
@@ -73,6 +75,32 @@ def self.safe_encode(str, encoding)
   require 'reline/unicode/east_asian_width'
 
   def self.get_mbchar_width(mbchar)
+    # If mbchar contains multiple characters, check if it's a valid combination
+    if mbchar.length >= 2 && invalid_combining_mark_cluster?(mbchar)
+      # Calculate width for each character separately for invalid combinations
+      mbchar.each_char.sum { |char| get_single_char_width(char) }
```

Comment:
considering multi-character grapheme cluster followed by halfwidth dakuten/handakuten, 
`"👨‍👩‍👦‍👦ﾞ"` (== `"\u{1f468}\u{200d}\u{1f469}\u{200d}\u{1f466}\u{200d}\u{1f466}\u{ff9e}"`), size should be 3.
```
\u{1f468}: width = 2 defined in unicode EastAsianWidth.txt
\u{200d}: width=1 but ignore it. (I think this is not strictly defined in unicode spec)
\u{1f469}
\u{200d}
\u{1f466}
\u{200d}
\u{1f466}
\u{ff9e}: additional 1 width for halfwidth dakuten/handakuten
```

Reply:
For this part, the original implementation doesn't looks so good, so either is fine.
For the fast path in get_mbchar_width, respecting the original implementation would be helpful.